mcmd/commands/add.py

```diff
@@ -130,3 +162,96 @@ def add_token(args):
+            'Adding bootstrap 3 theme {} to bootstrap themes'.format(
+                highlight(bs3_name)))
+    if not args.from_path:
+        paths = [_get_path_from_quick_folders(theme) for theme in paths]
+    files = _prepare_files_for_upload(paths, names, valid_types)
+    post_files(files, api)
+
+
+@login
+def add_logo(args):
+    """
+    add_logo uploads a logo to add to the left top of the menu
+    :param args: commandline arguments containing path to logo
+    :return: None
+    """
+    io.start('Adding logo from path {}'.format(highlight(args.logo)))
+    api = config.api('logo')
+    valid_types = {'image/jpeg', 'image/png', 'image/gif'}
+    logo = [args.logo]
+    if not args.from_path:
+        logo = [_get_path_from_quick_folders(args.logo)]
+    files = _prepare_files_for_upload(logo, ['logo'], valid_types)
+    post_files(files, api)
+
+
+def _prepare_files_for_upload(paths, names, valid_content_types):
+    """
+    _prepare_files_for_upload takes the paths to the files to upload, the names of them and a list of valid content
+    types to translate to content types and generates a dictionary which can be uploaded in a post request
+    :param paths: a list of paths to the files to upload
+    :param names: the names of files to upload
+    :param valid_content_types: set of the possible valid content types
+    :return: a dictionary with as key the name of the file and as value a tuple with: filename, file to upload, and
+    content type
+
+    :exception McmdError when the file on the given path does not exist and when the content type of the file is invalid.
+    """
+    files = {}
+    for name, path in zip(names, paths):
+        file_name = get_file_name_from_path(path)
+        content_type = mimetypes.guess_type(path)[0]
+        if not os_path.exists(path):
+            raise McmdError(
+                'File [{}] does not exist on path [{}]'.format(file_name, path.strip(file_name)))
+        elif content_type in valid_content_types:
+            try:
+                files[name] = (file_name, open(path, 'rb'), content_type)
+            except FileNotFoundError:
+                raise McmdError(
+                    'File [{}] does not exist on path [{}]'.format(file_name, path.strip(file_name)))
+        else:
+            raise McmdError(
+                'File [{}] does not have valid content type [{}], content type should be in {}'.format(file_name,
+                                                                                                       content_type,
+                                                                                                       valid_content_types))
+    return files
+
+
+def _get_resource_folders():
+    return [Path(folder) for folder in config.get('resources', 'resource_folders')]
+
+
+def _get_path_from_quick_folders(file_name):
+    file_name = os_path.splitext(file_name)[0]
+    file_map = scan_folders_for_files(config.git_paths() + _get_resource_folders())
```

Comment:
It's not needed to look in the git folders I think since they all point to folders with test datasets.